astropy/cosmology/core.py

```diff
@@ -2562,6 +2867,13 @@ def _float_or_none(x, digits=3):
     fmtstr = "{0:.{digits}g}".format(x, digits=digits)
     return fmtstr.format(x)
 
+def vectorize_if_needed(func, x):
+    """ Helper function to vectorize functions on array inputs"""
+    if not isiterable(x):
+        return func(x)
+    else:
+        return np.vectorize(func)(x)
+
```

Comment:
Unfortunately that doesn't work, because you can't guarantee that func will fail with a `TypeError`.  For example, if try this change, `scipy.integrate.quad` fails with `ValueError`s.  A lot. And I'm leery of just catching any exception.